src/python_testing/TC_AVSM_2_1.py

```diff
@@ -161,276 +164,443 @@ async def test_TC_AVSM_2_1(self):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.AllocatedSnapshotStreams
             )
             logger.info(f"Rx'd AllocatedSnapshotStreams: {value}")
+            asserts.assert_is_not_none(value, "AllocatedSnapshotStreams is None")
+            # TODO assert struct fields of list - currently list is empty as nothing allocated
 
         self.step(20)
         if self.pics_guard(self.check_pics("AVSM.S.A0012")):
-            value = await self.read_single_attribute_check_success(
+            streamUsagePriorities = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.StreamUsagePriorities
             )
-            logger.info(f"Rx'd StreamUsagePrioritiesList: {value}")
+            logger.info(f"Rx'd StreamUsagePrioritiesList: {streamUsagePriorities}")
+            matter_asserts.assert_all(
+                streamUsagePriorities,
+                lambda x: x < Clusters.Globals.Enums.StreamUsageEnum.kUnknownEnumValue,
+                "StreamUsage is not a valid StreamUsageEnum",
+            )
 
         self.step(21)
         if self.pics_guard(self.check_pics("AVSM.S.A0013")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.SoftRecordingPrivacyModeEnabled
             )
             logger.info(f"Rx'd SoftRecordingPrivacyModeEnabled: {value}")
+            asserts.assert_is_not_none(value, "SoftRecordingPrivacyModeEnabled is None")
 
         self.step(22)
         if self.pics_guard(self.check_pics("AVSM.S.A0014")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.SoftLivestreamPrivacyModeEnabled
             )
             logger.info(f"Rx'd SoftLivestreamPrivacyModeEnabled: {value}")
+            asserts.assert_is_not_none(value, "SoftLivestreamPrivacyModeEnabled is None")
 
         self.step(23)
         if self.pics_guard(self.check_pics("AVSM.S.A0015")):
             value = await self.read_single_attribute_check_success(
                 endpoint=endpoint, cluster=cluster, attribute=attr.HardPrivacyModeOn
             )
             logger.info(f"Rx'd HardPrivacyModeOn: {value}")
+            asserts.assert_is_not_none(value, "HardPrivacyModeOn is None")
 
         self.step(24)
         if self.pics_guard(self.check_pics("AVSM.S.A0016")):
             value = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attr.NightVision)
             logger.info(f"Rx'd NightVision: {value}")
+            asserts.assert_is_not_none(value, "NightVision is None")
+            asserts.assert_less(
+                value, cluster.Enums.TriStateAutoEnum.kUnknownEnumValue, "NightVision is not a valid TriStateAutoEnum"
+            )
```

Comment:
The correct method for all enum assertions:

`asserts.assert_in(value, cluster.Enums.TristateAutoEnum, "some error")` 

This properly validates enums with gaps and is less brittle.

Ex:

```
>>> from enum import IntEnum
>>> class SomeEnum(IntEnum):
...     kFirst = 1
...     kThird = 3
... 
>>> a = 1
>>> a in SomeEnum
True
>>> a = 2
>>> a in SomeEnum
False
>>> 
```

Reply:
Thank you for the review. We have fixed it.